### PR TITLE
Add informative Gravatar UI to the profile editing

### LINKED ISSUE #20173: Task: Add informative Gravatar UI to the profile editing
As part of the recent Gravatar efforts we are adding an informative card to the profile editing. Because currently users don't know that their profile info is linked to their Gravatar profile and making changes here also makes changes there. This is also a way to introduce Gravatar to users.

Account 39

### PR DESCRIPTION
**Pull request by hamorillo**
Fixes #20173

-----

We want to inform users that when editing a profile, they are actually editing their Gravatar profile. In this PR, we are adding an informative view for that.

We've created a similar [PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/22563) on the iOS repo to keep both apps on a similar page.

The difference with the iOS app is that we'll show the Gravatar information banner in `MeFragment` instead of `MyProfileFragment`. That's because, in the Android app, users modify their avatar on the first screen, so we need to show the notice banner on that screen. 

Sync info banner will be shown in both fragments after any modification (avatar or field) is applied.

Jetpack:

| Dark Mode | Light Mode | Dark Mode Sync Message | Light Mode Sync Message | Dark - Profile Sync | Light - Profile Sync |
|----------|-----------|----------|-----------|----------|-----------|
|image|image|image|image|image|image|

WordPress:

| Dark Mode | Light Mode | Dark Mode Sync Message | Light Mode Sync Message | Dark - Profile Sync | Light - Profile Sync |
|----------|-----------|----------|-----------|----------|-----------|
|image|image|image|image|image|image|

## To Test:

Try on both JP and WP:

- Go to the `Me tab`
- Observe: There's a new informative UI as demonstrated above ☝️ (_If possible, verify in dark and light mode_)
- Click the link `Learn more on Gravatar.com` and verify you are redirected to `www.gravatar.com`
- Go back to the App
- Update your avatar
- Observe you see a notice message saying "Updates might take some time to sync with your Gravatar profile."
- Click the `Done` button and verify the notice message disappears
- Open `My Profile`
- Try updating those 4 fields
- Observe you see a notice message saying "Updates might take some time to sync with your Gravatar profile."
- Click the `Done` button and verify the notice message disappears

-----

## Regression Notes

1. Potential unintended areas of impact

    - We shouldn't impact other areas. We are adding views in `MyProfileFragment` and `MeFragment`, so all possible impacts should be inside that fragment.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - N/A

3. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist:

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by Luchadores** [comment]
Hello, these look good too, I just wonder why is the Jetpack background fully white on White mode. For me, the screenshots I took look different where we have a slight gray background from the app itself. I will attach the image below for reference. 

image

**Comment by hamorillo** [comment]
> Hello, these look good too, I just wonder why is the Jetpack background fully white on White mode. For me, the screenshots I took look different where we have a slight gray background from the app itself. I will attach the image below for reference.

@Luchadores If I'm not wrong, your screenshots were from the iOS app. The Android app has a fully white background, but I'm not sure why that difference, though. Anyway, I've used the already defined values in the app for the card background, so It should be good.

**Comment by maxme** [comment]
I know it's not related to your code Hector, but the link to gravatar.com landing page is not a good experience:
- Mobile web margins seems broken (buttons are truncated).
- GDPR notice is truncated and the buttons are not visible.

https://github.com/wordpress-mobile/WordPress-Android/assets/40213/e4e9566a-f9e4-4c0f-9d0d-bb4633429001


---


I tried to find alternatives to the current link, but I couldn't find any better option. One alternative would be the "What's a Gravatar?" page from the FAQ but mobile web styling isn't really good either.




Conclusion to this: I think we should keep the link as it and fix the mobile web experience on gravatar.com. I'll open a ticket for that. cc @Luchadores

**Comment by Luchadores** [comment]
Good point, @maxme. I agree, we should lead them to the same link, gravatar.com, and fix the mobile experience in a separate project/ticket.

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -2869,6 +2869,10 @@
     <string name="premium_upgrades_message">You have active premium upgrades on your site. Please cancel your upgrades prior to deleting your site.</string>
     <string name="show_purchases">Show purchases</string>
     <string name="checking_purchases">Checking purchases</string>
+    <string name="gravatar_info_description">"Gravatar keeps your profile information safe and up to date, automatically syncing any updates made here with your Gravatar profile."</string>
```
Comment (ravishanker) [inline WordPress/src/main/res/values/strings.xml]: What are the quotes (") for in this string?

**Review by ravishanker — APPROVED** [review]
LGTM 🚀

**Comment by ravishanker** [comment]
The english version of https://gravatar.com/ seems to be OK, probably other languages have issue on mobile web layout.

File: WordPress/src/main/res/values/strings.xml (hunk 1 above)
Comment (hamorillo) [reply WordPress/src/main/res/values/strings.xml]: Good catch. Removed! Thanks!

**Review by maxme — CHANGES_REQUESTED** [review]
Tested with the following contexts, everything looks good:
- [x] WordPress.com sites and self-hosted Jetpack sites.
- [x] Light and dark modes.
- [x] High contrast.
- [x] Talkback.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)

There is an issue since the notice is not scrollable and can't be hidden, the profile fields are hidden and unusable. With the following contexts:
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)
- [ ] Portrait and landscape orientations.
- [ ] Fonts: Larger, smaller and bold text.

## landscape


## split screen


## bigger text size

**Comment by hamorillo** [comment]
> There is an issue since the notice is not scrollable and can't be hidden, the profile fields are hidden and unusable. 

❤️ Great catch @maxme! I've fixed the constraint of `nested_scroll_view` but there are some corner cases where the UX can still be improved. However, for those cases, I think we need to talk with designers about the possibility of making the notice hideable. Not sure if those are blocking, though. Wdyt?

image

**Review by maxme — APPROVED** [review]
> I've fixed the constraint of nested_scroll_view but there are some corner cases where the UX can still be improved.

Works fine, the view is not ideal when hidden behind the notice but still usable, so :shipit: 

> Not sure if those are blocking, though. Wdyt?

Not blocking IMO there are parts of the app that won't work when you're in landscape + huge text.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>strings.xml</code> files should only be updated on release branches, when the translations are downloaded by our automation.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-Android/milestone/277">24.3</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by hamorillo** [comment]
@ravishanker @maxme, Sorry for bothering you again. I've moved the Gravatar info banner from `MyProfileFragment` to `MeFragment`. I've also updated the PR description. Can you do another round?

**Comment by maxme** [comment]
Code and UI looks good to me, but now I have a doubt about the copy, specifically the mention to "WordPress.com". When I run the Jetpack variant: 

Screenshot_20240216_144359

@ravishanker maybe you can help, is this correct to mention "WordPress.com profile"on the Jetpack variant? Should we change the copy? I think it's fine since the login screens mention a "WordPress.com account" as well:

Screenshot_20240216_145504

**Comment by Luchadores** [comment]
Uuu, good catch. Could you tag Ronnie, I cannot seem to find his handle here. @maxme

**Comment by ravishanker** [comment]
> is this correct to mention "WordPress.com profile"on the Jetpack variant

Yes, that's correct.  It is a [WordPress.com](https://wordpress.com/) account

If anything, the `About Jetpack`, `Log out of Jetpack` should probably just read `About`, and `Logout`.  Not for this PR though.

**Comment by ravishanker** [comment]
Me screen is a better location than profile screen.  Debug Settings, and Design System options there won't appear on Release version, so it should appear in full above the bottom nav.  However
- The background contrast is now much lighter now than before in both light, and dark theme.  Is that intentional?  
- The title (Your WordPress.com....) font is too bold, and shouting, feels out of place with the rest of the app.  Look at My Site and else where for reference

**Comment by maxme** [comment]
> Me screen is a better location than profile screen. Debug Settings, and Design System options there won't appear on Release version, so it should appear in full above the bottom nav. However
> 
>   - The background contrast is now much lighter now than before in both light, and dark theme.  Is that intentional?
> 
>   - The title (Your WordPress.com....) font is too bold, and shouting, feels out of place with the rest of the app.  Look at My Site and else where for reference

@Luchadores what do you recommend with the title font weight and the background color?

**Comment by maxme** [comment]
Updated with a smaller text size for the title

Light|Dark
-|-
Screenshot_20240219_093114|Screenshot_20240219_093054

**Comment by maxme** [comment]
Darker background `#F2F2F7` for light theme:



Dark theme was already using `#1C1C1E`, not sure if you want to change this one.

**Comment by mlumeau** [comment]
> Updated with a smaller text size for the title

I think it's a bit better, minus the orphan ("Gravatar" alone in a new line).
> Darker background #F2F2F7 for light theme

I think it's much better against the Android app's white background
cc: @Luchadores

**Comment by Luchadores** [comment]
👋 

Thank you all for looking at this, it's much appreciated. I think colors are fine, as we have to have a bit of contrast between the page background and the notice after updating your profile. 

We can also go with the text size from Maxime if that fits better. I don't have an Android phone, so I always have to ask for screenshots of existing UI.

PS: I would force a line break so that we don't end up having orphans like we have in the screenshot with "Gravatar"
image

Thank you!
@mlumeau @maxme

**Comment by maxme** [comment]
We don't need to force a line break (this would look weird in landscape mode or on a tablet). I fixed it throught WPTextView in 1e331558d7940e6d431aef4ff14aa1ee0e943c7a, it replaces the last space of a sentence by a non breakable space.

**Review by maxme — APPROVED** [review]

**Comment by mlumeau** [comment]
> I fixed it throught WPTextView in https://github.com/wordpress-mobile/WordPress-Android/commit/1e331558d7940e6d431aef4ff14aa1ee0e943c7a, it replaces the last space of a sentence by a non breakable space.

TIL!

**Pull request merged**